### PR TITLE
Clippy Fix 4

### LINKED ISSUE #50: Clippy Fix 4
Fix following clippy warning
```
warning: method is never used: `fetch_slice`
  --> async_fuse/src/fuse_request.rs:52:5
   |
52 |     pub fn fetch_slice<T>(&mut self) -> anyhow::Result<Vec<&'a T>> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: this function has too many arguments (8/7)
   --> fuse_ll/src/fuse/mod.rs:306:5
    |
306 | /     fn release(
307 | |         &mut self,
308 | |         _req: &Request<'_>,
309 | |         _ino: u64,
...   |
314 | |         reply: ReplyEmpty,
315 | |     ) {
    | |______^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: method is never used: `uid`
   --> async_fuse/src/fuse_request.rs:698:5
    |
698 |     pub fn uid(&self) -> u32 {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `gid`
   --> async_fuse/src/fuse_request.rs:704:5
    |
704 |     pub fn gid(&self) -> u32 {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `pid`
   --> async_fuse/src/fuse_request.rs:710:5
    |
710 |     pub fn pid(&self) -> u32 {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `len`
   --> async_fuse/src/fuse_request.rs:716:5
    |
716 |     pub fn len(&self) -> u32 {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: this function has too many arguments (8/7)
   --> fuse_ll/src/fuse/mod.rs:396:5
    |
396 | /     fn setxattr(
397 | |         &mut self,
398 | |         _req: &Request<'_>,
399 | |         _ino: u64,
...   |
404 | |         reply: ReplyEmpty,
405 | |     ) {
    | |______^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: constant item is never used: `FOPEN_DIRECT_IO`
   --> async_fuse/src/protocal.rs:152:1
    |
152 | pub const FOPEN_DIRECT_IO: u32 = 1 << 0;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: constant item is never used: `FOPEN_KEEP_CACHE`
   --> async_fuse/src/protocal.rs:153:1
    |
153 | pub const FOPEN_KEEP_CACHE: u32 = 1 << 1;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: constant item is never used: `FUSE_POSIX_LOCKS`
   --> async_fuse/src/protocal.rs:195:1
    |
195 | pub const FUSE_POSIX_LOCKS: u32 = 1 << 1;
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by Nugine**
Ignore unused codes.
Skip to fix too many arguments.

### CODE REVIEW

**Comment by rogercloud** [comment]
Where's the fix for "too many arguments"?

**Comment by Nugine** [comment]
> Where's the fix for "too many arguments"?

The warnings belong to a huge trait `FileSystem`:

<https://github.com/datenlord/datenlord/blob/e74f6f6f677aa669aa4dac8c2fbf1bb184ec79a2/fuse_ll/src/fuse/mod.rs#L98-L579>

I'm not sure how to fix the problem without changes on design. So I just skipped like what #62 did.

File: async_fuse/src/protocol.rs [hunk 1]
```diff
@@ -149,7 +149,9 @@ pub const FATTR_FLAGS: u32 = 1 << 31;
 // FOPEN_CACHE_DIR: allow caching this directory
 // FOPEN_STREAM: the file is stream-like (no file position at all)
 //
+#[allow(dead_code)]
```
Comment (pwang7) [inline async_fuse/src/protocol.rs:152]: for the const variables, maybe it's better to use a mod to group them, and add allow dead_code to the mod.

Comment (Nugine) [reply async_fuse/src/protocol.rs]: Done in <https://github.com/datenlord/datenlord/pull/68/commits/f1168fc807b1e96c2d0db1b04c8122dece7b8e44>

**Comment by rogercloud** [comment]
Please add the related issue number to the commit header, keeping the commit easy to track.

File: async_fuse/src/protocol.rs [hunk 2]
```diff
@@ -141,26 +141,50 @@ pub const FATTR_BKUPTIME: u32 = 1 << 30;
...
+    //!
+    //! FOPEN_NONSEEKABLE: the file is not seekable
+    //!
+    //! FOPEN_CACHE_DIR: allow caching this directory
+    //!
+    //! FOPEN_STREAM: the file is stream-like (no file position at all)
+    //!
+
+    /// bypass page cache for this open file
+    pub const FOPEN_DIRECT_IO: u32 = 1 << 0;
```
Comment (pwang7) [inline async_fuse/src/protocol.rs:161]: BTW, change 1 << 0 to 1

File: async_fuse/src/protocol.rs [hunk 3]
```diff
@@ -141,26 +141,50 @@ pub const FATTR_BKUPTIME: u32 = 1 << 30;
...
-#[cfg(feature = "abi-7-28")]
-pub const FOPEN_CACHE_DIR: u32 = 1 << 3;
-#[cfg(feature = "abi-7-31")]
-pub const FOPEN_STREAM: u32 = 1 << 4;
-#[cfg(target_os = "macos")]
-pub const FOPEN_PURGE_ATTR: u32 = 1 << 30;
-#[cfg(target_os = "macos")]
-pub const FOPEN_PURGE_UBC: u32 = 1 << 31;
+pub mod fopen_flags {
+    #![allow(dead_code)]
```
Comment (pwang7) [inline async_fuse/src/protocol.rs:145]: how about this:
```
#[allow(dead_code)]
pub mod fopen_flags {
```

Comment (Nugine) [reply async_fuse/src/protocol.rs]: done

File: fuse_ll/src/fuse/mod.rs [hunk 4]
```diff
@@ -95,6 +95,7 @@ pub struct FileAttr {
 /// These methods correspond to fuse_lowlevel_ops in libfuse. Reasonable default
 /// implementations are provided here to get a mountable filesystem that does
 /// nothing.
+#[allow(clippy::too_many_arguments)]
```
Comment (pwang7) [inline fuse_ll/src/fuse/mod.rs:98]: don't allow this lint, to fix it, use a struct to group function arguments.

File: async_fuse/src/protocol.rs [hunk 5]
```diff
@@ -192,6 +215,7 @@ pub const FOPEN_PURGE_UBC: u32 = 1 << 31;
 // FUSE_EXPLICIT_INVAL_DATA: only invalidate cached pages on explicit request
 //
 pub const FUSE_ASYNC_READ: u32 = 1 << 0;
+#[allow(dead_code)]
```
Comment (pwang7) [inline async_fuse/src/protocol.rs:218]: also group these const variables to a mod

**Comment by Nugine** [comment]
ready to merge

**Pull request merged**